### PR TITLE
Implementation for commandSetIterator function.

### PR DESCRIPTION
**Pull request by samwondim**

### CODE REVIEW

**Review by Yagth — CHANGES_REQUESTED** [review]

**Comment by samwondim** [comment]
@Yagth  The issues are fixed.

File: enf-reducer/RTEHelpers.metta [hunk 1]
```diff
@@ -33,6 +33,32 @@
...
+
+;; -----------------------------------
+;; -----------------------------------
+;; -------- A function that given children and the a local command set,
+;; -------- it returns the updated command set
+;; -----------------------------------
+;; -----------------------------------
+
+(: commandSetIterator(-> (List Tree) (List Tree) (List Tree)))
+(= (commandSetIterator $children $localCommandSet)(
```
[the remark this answers is not in the document]
Comment (Yagth) [reply enf-reducer/RTEHelpers.metta:53]: Better but this still can be better. You can also define the function with different cases with out using the `case` key word. like so
```
(= (commandSetIterator Nil $localCommandSet) $localCommandSet)
(= (commandSetIterator (Cons $x $xs) $localCommandSet 
    (...)
)
```

**Review by Yagth — COMMENTED** [review]
You can still make the `commandSetIterator` function better without even using case.

File: enf-reducer/RTEHelpers.metta [hunk 2]
```diff
@@ -33,6 +33,32 @@
...
+  if(== $children Nil)($localCommandSet) (
+      if(and (and (== (getChildren (head $children)) Nil) (== (guardSetChecker $children) 1))
+            (and (not (== $children Nil)) (== (getNodeType (head $children)) AND))
+        )(
+          setUnion (getGuardSet (head $children)) (commandSetIterator (tail $children) $localCommandSet)
```
[the remark this answers is not in the document]
Comment (Yagth) [reply enf-reducer/RTEHelpers.metta]: Isn't the `setUnion` function expecting a `treeIsMember` function here?

**Review by Yagth — APPROVED** [review]

**Pull request merged**